### PR TITLE
refactor: replace NavigationManager with Navigator - Part 3 [WPB-384]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-384" title="WPB-384" target="_blank">Sub-taskWPB-384</a>  implement navigator (refactor NavigationManager)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Our current NavigationManager was supposed to make navigation easier, but it has been added to many places that in theory shouldn't make any navigation actions directly. Navigation should be executed in the UI composables, especially when we use Compose Destinations which alraedy forced us to move back navigation with arguments to composables, so to not make it even more complicated, we should decide on one unified way of navigating, so it comes down to moving the navigation to composables.

### Solutions

In this PR, next part of composable screens and their ViewModels are refactored to use Navigator instead of NavigationManager, mainly ones related to logging in, migration, settings and calling.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Just navigate, go to one of the refactored screens.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1883/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1883/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6856289)(6856289)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6856289)(6856289)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1883/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6880831)(6880831)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6880831)(6880831)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailState.kt [hunk 1]
```diff
@@ -21,17 +21,25 @@ import androidx.compose.ui.text.input.TextFieldValue
...
-            object InvalidEmail : TextFieldError
-            object Generic : TextFieldError
+
+    sealed interface FlowState {
+        object Default : FlowState
+        object Loading : FlowState
+        data class Success(val newEmail: String) : FlowState
+        object NoChange : FlowState
+        sealed interface Error : FlowState {
+            object None : Error
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailState.kt]: Strange state inside ErrorState 🤔

**Review by Garzas — APPROVED** [review]
Never ending refactoring 🚀 Just one question about one state 😄

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailState.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailState.kt]: Right, it's not used, just forgot to remove 😅 Thanks!

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1883/5/) **failed**.

**Review by trOnk12 — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1883/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6882645)(6882645)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6882645)(6882645)-staging-compat.apk)

**Review by yamilmedina — APPROVED** [review]
Titanic work here, eager to see the final result 👏

**Pull request merged**